lib/net.js

```diff
@@ -935,12 +958,16 @@ Socket.prototype.connect = function(options, cb) {
 Socket.prototype.ref = function() {
   if (this._handle)
     this._handle.ref();
+  else
+    this._handleOptions.ref = true;
```

Comment:
I think this `ref`  should actually be an int and we should `++` and `--` it accordingly.